File: fixtures/scale/csharp/medium/Repl/InputReader.cs
using System;

namespace Calc.Repl
{
    public class InputReader
    {
        private readonly string _prompt;

        public InputReader(string prompt)
        {
            _prompt = prompt;
        }

        public string NextLine()
        {
            Console.Write(_prompt);
            return Console.In.ReadLine();
        }
    }
}

File: fixtures/scale/java/medium/src/calc/repl/InputReader.java
package calc.repl;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

public class InputReader {
    private final BufferedReader reader;

    public InputReader(InputStream in) {
        this.reader = new BufferedReader(new InputStreamReader(in));
    }

    public String nextLine() {
        try {
            return reader.readLine();
        } catch (IOException e) {
            return null;
        }
    }
}
